scanpy/preprocessing/_simple.py

```diff
@@ -148,11 +148,16 @@ def filter_cells(
     X = data  # proceed with processing the data matrix
     min_number = min_counts if min_genes is None else min_genes
     max_number = max_counts if max_genes is None else max_genes
-    number_per_cell = np.sum(
-        X if min_genes is None and max_genes is None else X > 0, axis=1
-    )
-    if issparse(X):
-        number_per_cell = number_per_cell.A1
+    if min_genes is None and max_genes is None:
+        number_per_cell = np.sum(X, axis=1)
+        if issparse(X):
+            number_per_cell = number_per_cell.A1
+    elif isspmatrix_csr(X):
+        number_per_cell = np.diff(X.indptr)
```

Comment:
I believe that this will work for `csr_array` (and `csr_matrix` of course) if you change this `elif` clause:
```suggestion
    elif issparse(X) and X.format == 'csr':
        number_per_cell = np.diff(X.indptr)
```